AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/utils/Utils.java

```diff
@@ -98,59 +101,95 @@ private static Toast getToast(String message, Context context, int length) {
       intent.setPackage(packageName);
-      //startActivity() might throw ActivityNotFoundException error
+      // startActivity() might throw ActivityNotFoundException error
       context.startActivity(intent);
     } catch (Exception e) {
       String errorMsg;
-      if(e instanceof URISyntaxException) {
+      if (e instanceof URISyntaxException) {
         errorMsg = context.getString(R.string.error_parsing);
-      }else if(e instanceof ActivityNotFoundException) {
+      } else if (e instanceof ActivityNotFoundException) {
         errorMsg = context.getString(R.string.error_activity);
-      }else{
+      } else {
         errorMsg = context.getString(R.string.error_unknown);
       }
       Utils.showMsg(errorMsg, context);
     }
   }
 
-
-  /**
-   * Dialog for user to choose among given list values.
-   */
-  public static void popUpDialog(final Context context, final String title, List<CharSequence> list, DialogInterface.OnClickListener listener) {
+  /** Dialog for user to choose among given list values. */
+  public static void popUpDialog(
+      final Context context,
+      final String title,
+      List<CharSequence> list,
+      DialogInterface.OnClickListener listener) {
     MaterialAlertDialogBuilder materialAlertDialogBuilder = new MaterialAlertDialogBuilder(context);
     materialAlertDialogBuilder.setTitle(title);
     CharSequence[] keys = new CharSequence[list.size()];
     materialAlertDialogBuilder.setItems(list.toArray(keys), listener).show();
   }
 
-  // Jump to a deep link which has no parameter or jump into parameterActivity to select parameters for the deeplink.
-  public static void jumpByType(final Context context, final AppActionProtos.AppAction appAction, final AppActionProtos.Action action, final AppActionProtos.FulfillmentOption fulfillmentOption) {
-    if (fulfillmentOption.getUrlTemplate().getTemplate().equals(Constant.URL_NO_LINK) || fulfillmentOption.getUrlTemplate().getParameterMapCount() > 0) {
+  // Jump to a deep link which has no parameter or jump into parameterActivity to select parameters
+  // for the deeplink.
+  public static void jumpByType(
+      final Context context,
+      final AppAction appAction,
+      final Action action,
+      final FulfillmentOption fulfillmentOption) {
+    if (fulfillmentOption.getUrlTemplate().getTemplate().equals(Constant.URL_NO_LINK)
+        || fulfillmentOption.getUrlTemplate().getParameterMapCount() > 0) {
       Intent intent = new Intent(context, ParameterActivity.class);
       intent.putExtra(Constant.FULFILLMENT_OPTION, fulfillmentOption);
       intent.putExtra(Constant.ACTION, action);
       intent.putExtra(Constant.APP_NAME, appAction);
       context.startActivity(intent);
     } else {
-      Utils.jumpToApp(context, fulfillmentOption.getUrlTemplate().getTemplate(), appAction.getPackageName());
+      String url = fulfillmentOption.getUrlTemplate().getTemplate();
+      if (appAction.getPackageName().equals("com.dunkinbrands.otgo")) {
```

Comment:
add some comment of why we need to handle dunkin deeplinks specially

Reply:
Sure, comments added.